### PR TITLE
Add Metrics monitoring tests

### PR DESCRIPTION
**Pull request by anishasthana**
https://polarion.engineering.redhat.com/polarion/redirect/project/OpenDataHub/workitem/question_mark/id=ODS-509
https://polarion.engineering.redhat.com/polarion/redirect/project/OpenDataHub/workitem/question_mark/id=ODS-510

### CODE REVIEW

**Review by lugi0 — COMMENTED** [review]
lgtm

**Comment by jgarciao** [comment]
Thanks @anishasthana for your contribution!

I've run this code here: https://opendatascience-jenkins-csb-rhods.apps.ocp4.prod.psi.redhat.com/job/rhods-ci-pr-test/160/console

```
==============================================================================
Test Existence of Prometheus Alerting Rules                           | FAIL |
Variable '${RHODS_PROMETHEUS_URL}' not found.
```

${RHODS_PROMETHEUS_URL} and RHODS_PROMETHEUS_TOKEN should be added in a private repo we manage. I'll take care of that.

```
[ ERROR ] Error in file '/home/jenkins/workspace/rhods-ci-pr-test/160/ods-ci/tests/Resources/Page/ODH/Prometheus/Prometheus.robot' on line 4: Importing library 'RequestsLibrary' failed: ModuleNotFoundError: No module named 'RequestsLibrary'
```

You need to add robotframework-requests as a dependency in this file: https://github.com/red-hat-data-services/ods-ci/blob/master/setup.py

File: tests/Resources/Page/ODH/Prometheus/Prometheus.robot [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+Run Query
+  [Arguments]  ${pm_url}  ${pm_token}  ${pm_query}
+  ${pm_headers}=       Create Dictionary  Authorization=Bearer ${pm_token}
+  ${resp}=       GET   url=${pm_url}/api/v1/query?query=${pm_query}   headers=${pm_headers}   verify=${False}
+  Status Should Be    200    ${resp}
+  Log To Console    ${resp.json()}
+  [Return]   ${resp}
+
+Get Rules
+  ${pm_headers}=       Create Dictionary  Authorization=Bearer ${pm_token}
```
Comment (jgarciao) [inline tests/Resources/Page/ODH/Prometheus/Prometheus.robot:16]: [Arguments] should be after the keyword name

File: tests/Resources/Page/ODH/Prometheus/Prometheus.robot [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    IF  '${rule_name}' == '${rule_group}'
+      ${rules_list_len}=  Get Length  ${rules_list}
+      Should Be True  ${rules_list_len} != 0
+      Return from keyword  ${TRUE}
+    END
+  END
+  Log To Console  ${rule_group} was not found in Prometheus rules
+  Fail
+
+Verify Rules
```
Comment (jgarciao) [inline tests/Resources/Page/ODH/Prometheus/Prometheus.robot:38]: This keyword could be renamed to:  `Rules Should Contain` (following the SeleniumLibrary naming) with the same parameters or create different keywords for each type:

```
Rules Should Contain Alert
Rules Should Contain Record
```

File: tests/Tests/200__monitor_and_manage/200__metrics/200__metrics.robot [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+*** Variables ***
+
+@{RECORD_GROUPS}  Availability Metrics  SLOs - JupyterHub  SLOs - JupyterHub DB  SLOs - ODH Dashboard  SLOs - RHODS Operator  SLOs - Traefik Proxy  Usage Metrics
+
+@{ALERT_GROUPS}  Builds  DeadManSnitch  RHODS-PVC-Usage  SLOs-haproxy_backend_http_responses_total  SLOs-probe_success
+
+*** Test Cases ***
+
+Test Existence of Prometheus Alerting Rules
+  [Tags]  Metrics  Alerting  Prometheus
```
Comment (jgarciao) [inline tests/Tests/200__monitor_and_manage/200__metrics/200__metrics.robot:13]: You can use this tags:
```
    [Tags]  Smoke  ODS-509
```

File: tests/Tests/200__monitor_and_manage/200__metrics/200__metrics.robot [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+*** Variables ***
+
+@{RECORD_GROUPS}  Availability Metrics  SLOs - JupyterHub  SLOs - JupyterHub DB  SLOs - ODH Dashboard  SLOs - RHODS Operator  SLOs - Traefik Proxy  Usage Metrics
+
+@{ALERT_GROUPS}  Builds  DeadManSnitch  RHODS-PVC-Usage  SLOs-haproxy_backend_http_responses_total  SLOs-probe_success
+
+*** Test Cases ***
+
+Test Existence of Prometheus Alerting Rules
+  [Tags]  Smoke  ODS-509
```
Comment (jgarciao) [inline tests/Tests/200__monitor_and_manage/200__metrics/200__metrics.robot:13]: Sorry, in the end it's better that this goes to Sanity first
```
 [Tags]  Sanity  ODS-509
```

File: tests/Tests/200__monitor_and_manage/200__metrics/200__metrics.robot [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+  Check Prometheus Alerting Rules
+
+Test Existence of Prometheus Recording Rules
+  [Tags]  Smoke  ODS-510
```
Comment (jgarciao) [inline tests/Tests/200__monitor_and_manage/200__metrics/200__metrics.robot:17]: Sorry, in the end it's better that this goes to Sanity first
```
 [Tags]  Sanity  ODS-510
```

File: tests/Tests/200__monitor_and_manage/200__metrics/200__metrics.robot (hunk 4 above)
Comment (anishasthana) [reply tests/Tests/200__monitor_and_manage/200__metrics/200__metrics.robot]: Fixed!

File: tests/Tests/200__monitor_and_manage/200__metrics/200__metrics.robot (hunk 5 above)
Comment (anishasthana) [reply tests/Tests/200__monitor_and_manage/200__metrics/200__metrics.robot]: Fixed!

**Comment by anishasthana** [comment]
@jgarciao Updated again :-)

**Comment by anishasthana** [comment]
This is waiting on #97

**Review by vasukulkarni — APPROVED** [review]

**Pull request merged**